test.rb

```diff
@@ -12,7 +13,7 @@
 
     describe "POST /products" do
       before (:each) do
-        @response = Faraday.post("http://#{address}:#{port}/products", $body)
+        @response = Faraday.post("http://#{address}:#{port}/products", JSON.generate($body))
```

Comment:
Content-Type è automatico o bisogna esplicitarlo? 
Faraday uses convention to assume that any content you're sending in a POST request is of the type application/x-www-form-urlencoded

A me funziona senza JSON.generate, però il mio backend è fatto in rails.. magari fa qualche magia

Reply:
probabilmente dev'essere esplicito ma non credo importi visto che jsonapi definisce tutta la comunicazione come json (quindi sia client che server possono assumere che sia sempre json anche se non specificato).

Anche il client non controlla se il content type è json:
https://github.com/cristianogregnanin/json_api_test/blob/de65534b345f474effbb858efc66b5b6176772a0/test.rb#L15-L16

https://github.com/cristianogregnanin/json_api_test/blob/de65534b345f474effbb858efc66b5b6176772a0/test.rb#L38-L39

https://github.com/cristianogregnanin/json_api_test/blob/de65534b345f474effbb858efc66b5b6176772a0/test.rb#L56-L57

Reply:
ok, quindi JSON.parse lo mettiamo o no? a me funziona anche senza. Per me meno cose si mettono meglio è

Reply:
non usare JSON.generate non invia json, invia un form urlencoded
usare JSON.generate senza cambiare gli header invia json con un content type errato
usare JSON.generate e indicare gli header esplicitamente invia json con il content type corretto

`@response = Faraday.post("http://#{address}:#{port}/products", JSON.generate($body), {'Content-Type'=>'application/json'})`
questa è la versione "corretta" ma non specificare il content type (quindi quello che ho inviato nella pr) non dovrebbe essere un problema

Reply:
@response = Faraday.post("http://#{address}:#{port}/products", JSON.generate($body), {'Content-Type'=>'application/json'}))    usiamo questa